tests/local/gc/gc-subtypes-invalid.wast

```diff
@@ -129,3 +129,17 @@
   )
   "type index out of bounds"
 )
+(assert_invalid
+  (module
+    (type $A (struct (field $vt (mut i32))))
+    (type $C (sub $A (struct (field $tv (mut i32)))))
+  )
+  "unknown field"
+)
```

Comment:
Is this invalid according to the spec interpreter? This relates to my comment above about not checking parent types I think, and it's something I'm not sure about because this is valid from a binary format perspective so I would naively expect that this would be a valid module.

Reply:
This is perhaps a stronger check than what is explicitly prescribed by the GC spec. 

But it makes perfect sense: the spec does insist on field types being the same at the same positions as in the parent struct. So, unless we want to introduce a field renaming "feature", we don't want to allow naming inherited fields differently.

Reply:
We may actually want an even stronger check: if the parent struct field is named, the inherited field has to be named (the same) too.

Reply:
AFAIK the text format doesn't layer more restrictions over the binary format. In that sense I don't think the above would be an error. That being said I don't know, so if you're also not sure I think it would be good to check with the upstream proposal

Reply:
> AFAIK the text format doesn't layer more restrictions over the binary format.

Even with this PR it doesn't, because names of fields are gone in binary. The proposed restriction is on the text format. 

Having said that, you're probably right: this is way out of scope of a bug fix PR.

Removing the restriction for now.